src/lib/utils/flatten.js

```diff
@@ -19,18 +19,32 @@
 // THE SOFTWARE.
 
 /**
- * Flattens a nested array into a single level array
+ * Flattens a nested array into a single level array,
+ * or a single value into an array with one value
  * @example flatten([[1, [2]], [3], 4]) => [1, 2, 3, 4]
+ * @example flatten(1) => [1]
  * @param {Array} array The array to flatten.
- * @return {Array} Returns the new flattened array.
+ * @param {Function} filter= - Optional predicate called on each `value` to
+ *   determine if it should be included (pushed onto) the resulting array.
+ * @param {Array} result=[] - Optional array to push value into
+ * @return {Array} Returns the new flattened array (new array or `result` if provided)
  */
-export function flatten(array, result = []) {
+export function flatten(array, {
+  filter = () => true,
+  result = []
+} = {}) {
+  array = Array.isArray(array) ? array : [array];
+  return flattenArray(array, {filter, result});
+}
+
+// Deep flattens an array. Helper to `flatten`, see its parameters
+function flattenArray(array, {filter, result} = {}) {
```

Comment:
Does the construction and deconstruction of argument objects have perf impact in an incursive function? Given it's an internal helper we don't have to have a pretty interface.